src/components/slider/slider.tsx

```diff
@@ -1428,16 +1428,14 @@ export class Slider
   }
 
   /**
-   * Returns a string representing the localized label value based on groupSeparator prop being on or off.
+   * Returns a string representing the localized label value based if the groupSeparator prop is parsed.
    *
    * @param value
    */
-  private determineGroupSeparator = (value): string => {
+  private determineGroupSeparator = (value: number): number | string => {
     const lang = this.globalAttributes["lang"] || document.documentElement.lang || "en";
-    if (value) {
-      return this.groupSeparator
-        ? localizeNumberString(value.toString(), lang, this.groupSeparator, this.numberingSystem)
-        : value;
-    }
+    return this.groupSeparator && value
```

Comment:
What if `value` is `0`? Should it localize it still?

Reply:
It should. Updated the code.

Reply:
And also, earlier it is required to parse `group-separator` boolean to localize the numbers in `slider`. With this fix `group-separator `is not required. 

cc: @benelan , @geospatialem